HACKING.md

```diff
@@ -354,24 +383,39 @@ git checkout -b <branchname>
 ```
 
 Make changes to the repository and commit the changes to your
-working branch. Push the changes to your forked `snapd` repository.
+local working branch.
+
+```sh
+git commit -sa -m "<feature/subject-prefix>: <brief-description-of-the-change>
+
+<more-elaborate-description-with-reasoning-about-the-changed>"
+``` 
+
+>If it's justified, one might consider several commits before pushing 
```

Comment:
I totally understand. I need your help to understand one thing then. Do we have certain guidelines that are specific to us?

For instance, PR caption has to have format: "module/feature: caption", and I actually failed on my PR because of it. Are there any other requirements like these? 
It's not mandatory for GitHub or git, but if we verify as part of PR, we need to state it clear here, or in some other dedicated document.

Reply:
yes we should also mention the PR title checker and probably also that commit messages should not have non-ASCII characters as their email addresses like the recent checker that we added for this, other than that no I don't think we have other guidelines to document